homeassistant/components/trace/utils.py

```diff
@@ -1,22 +1,26 @@
 """Helpers for script and automation tracing and debugging."""
 from collections import OrderedDict
+from typing import Any, TypeVar
 
+_KT = TypeVar("_KT")  # key type
+_VT = TypeVar("_VT")  # value type
 
-class LimitedSizeDict(OrderedDict):
+
+class LimitedSizeDict(OrderedDict[_KT, _VT]):
     """OrderedDict limited in size."""
 
-    def __init__(self, *args, **kwds):
+    def __init__(self, *args: Any, **kwds: Any) -> None:
         """Initialize OrderedDict limited in size."""
         self.size_limit = kwds.pop("size_limit", None)
-        OrderedDict.__init__(self, *args, **kwds)
+        OrderedDict.__init__(self, *args, **kwds)  # type: ignore[arg-type]
```

Comment:
I think this is a bug in `mypy`, linked to generic+inheritance

Reply:
Try using `super()` instead.
```suggestion
        super().__init__(*args, **kwds)
```